File: 1518. Water Bottles/numWaterBottles.cpp
#include <iostream>
#include <algorithm>

using namespace std;

class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int ans = numBottles;

        while(numBottles >= numExchange){
            ans += floor(numBottles / numExchange);
            numBottles = floor(numBottles / numExchange) + (numBottles % numExchange);
        }


        return ans;
    }
};

File: 1518. Water Bottles/numWaterBottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function (numBottles, numExchange) {

    let ans = numBottles

    while (numBottles >= numExchange) {
        ans += Math.floor(numBottles / numExchange)
        numBottles = Math.floor(numBottles / numExchange) + (numBottles % numExchange)
    }

    return ans;
};